### PR TITLE
[Refactor] bottom-up data transforms

### PR DESCRIPTION
**Pull request by ly015**
## Motivation

Refactor bottom-up data transforms.

## Modification

1. Add `mmpose/datasets/transforms/bottomup_transforms.py`:

| refactored | original |
| --- | --- |
| `BottomupRandomAffine` | `BottomUpRandomAffine` (except mask) |
| `BottomupGetHeatmapMask` | `BottomUpRandomAffine` (only mask) |
| `BottomupResize` | `BottomUpGetImgSize` and `BottomUpResizeAlign` |

2. `TopdownGenerateTarget` -> `GenerateTarget`

This transform is moved from `topdown_transforms.py` to 'common_transforms.py' for general use in both top-down and bottom-up pipelines.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpose/codecs/wrappers.py [hunk 1]
```diff
@@ -1,99 +0,0 @@
...
-
-        assert isinstance(encoders, list) and len(encoders) > 1, (
-            '``encoders`` should be a list of more than 1 configs')
-
-        self.encoders = [KEYPOINT_CODECS.build(cfg) for cfg in encoders]
-
-        self.input_size = self.encoders[0].input_size
-        self.heatmap_size = self.encoders[0].heatmap_size
-
-        # check all encoders have the samle ``input_size`` and
```
Comment (Tau-J) [inline mmpose/codecs/wrappers.py:41]: typo: samle - same

File: mmpose/datasets/transforms/common_transforms.py [hunk 2]
```diff
@@ -860,3 +864,129 @@ def __repr__(self) -> str:
...
+
+    Args:
+        encoder (dict | list[dict]): The codec config for keypoint encoding
+        target_type (str): The type of the encoded form of the keypoints.
+            Should be one of the following options:
+
+            - ``'heatmap'``: The encoded should be instance-irrelevant
+                heatmaps and will be stored in ``results['heatmaps']``
+            - ``'multiscale_heatmap'`` The encoded should be a list of
+                heatmaps and will be stored in ``results['hatmaps']``. Note
```
Comment (Tau-J) [inline mmpose/datasets/transforms/common_transforms.py:897]: type: hatmaps - heatmaps

File: mmpose/datasets/transforms/common_transforms.py [hunk 3]
```diff
@@ -197,14 +201,18 @@ def transform(self, results: dict) -> dict:
             results['flip_direction'] = flip_dir
 
             h, w = results['img_shape']
-            # flip image
+            # flip image and make
```
Comment (jin-s13) [inline mmpose/datasets/transforms/common_transforms.py:204]: mask?

File: mmpose/structures/keypoint/transforms.py [hunk 4]
```diff
@@ -23,10 +23,12 @@ def flip_keypoints(keypoints: np.ndarray,
...
             unavailable
         image_size (tuple): The image shape in [w, h]
-        flip_pairs (list[tuple]): The list of symmetric keypoint pairs'
-            indices
+        flip_indices (List[int]): The indices of each keypoint's symmetric
+            keypoint
         direction (str): The flip direction. Options are ``'horizontal'``,
             ``'vertical'`` and ``'diagonal'``. Defaults to ``'horizontal'``
+        inplace (bool): Whether modify ``keypoints`` and ``keypoints_visible``
+            in-place. Defaults to ``False``
```
Comment (Tau-J) [inline mmpose/structures/keypoint/transforms.py:31]: what does the param `inplace` use for? If only use for overwriting `results` in `RandomFlip`, I don't think it's neccessary.

File: mmpose/datasets/transforms/common_transforms.py (hunk 2 above)
Comment (ly015) [reply mmpose/datasets/transforms/common_transforms.py:897]: fixed

File: mmpose/codecs/wrappers.py (hunk 1 above)
Comment (ly015) [reply mmpose/codecs/wrappers.py:41]: This file is deleted. Multi-scale heatmaps will be directly generated by the data transform `GenerateTarget`

File: mmpose/structures/keypoint/transforms.py (hunk 4 above)
Comment (ly015) [reply mmpose/structures/keypoint/transforms.py:31]: Added by mistakes, will remove it.

**Review by Tau-J — APPROVED** [review]

**Pull request merged**